src/interface/sub_98199c.c

```diff
@@ -0,0 +1,158 @@
+#include "../world/footpath.h"
+#include "../addresses.h"
+
+/**
+ *
+ *  rct2: 0x006874B0, 0x00687618, 0x0068778C, 0x00687902, 0x0098199C
+ *
+ * @param image_id (ebx)
+ * @param x_offset (al)
+ * @param y_offset (cl)
+ * @param bound_box_length_x (di)
+ * @param bound_box_length_y (si)
+ * @param bound_box_length_z (ah)
+ * @param z_offset (dx)
+ * @param bound_box_offset_x (0x009DEA52)
+ * @param bound_box_offset_y (0x009DEA54)
+ * @param bound_box_offset_z (0x009DEA56)
+ * @param rotation (ebp)
+ * @return (!CF) success
+ */
+bool sub_98199C(
+	uint32 image_id,
+	sint8 x_offset, sint8 y_offset,
+	sint16 bound_box_length_x, sint16 bound_box_length_y, sint8 bound_box_length_z,
+	uint16 z_offset,
+	sint16 bound_box_offset_x, uint16 bound_box_offset_y, sint16 bound_box_offset_z,
+	uint32 rotation
+) {
+	assert((uint16) bound_box_length_x == (sint16) bound_box_length_x);
+	assert((uint16) bound_box_length_y == (sint16) bound_box_length_y);
+	assert((uint16) z_offset == (sint16) z_offset);
+
+	if (RCT2_GLOBAL(0xF1AD28, uint32) == 0) {
```

Comment:
If there is no previous paint struct then there is nothing to attach var_20 onto so it does a 98197C